### PR TITLE
sci-electronics/kicad: Add new 4.0.5 ebuild, move to EAPI6

### PR DESCRIPTION
**Pull request by zpuskas**
Gentoo Bug: #603514, #576732

Package-Manager: Portage-2.3.3, Repoman-2.3.1

### CODE REVIEW

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	xdg_src_prepare
+
+	# Add separated out libcontext files and patch source to use them
+	mkdir -p "${S}/common/system/" || die
+	mkdir -p "${S}/include/system/" || die
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.cpp" "${S}/common/system/libcontext.cpp" || die
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.h" "${S}/include/system/libcontext.h" || die
+	# Path source to use new "built in" libcontext. Also patch libcontext.cpp to have correct include file.
+	# Path must be applied after new libcontext files have been copied to the kicad source directory.
+	epatch "${FILESDIR}/${PN}-4.0.4-boost-context.patch"
```
Comment (gktrk) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: I know you need to inherit eutils for edos2unix anyway, but it'd be better if we switch to eapply here

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+for lang in ${LANGS} ; do
+	IUSE="${IUSE} linguas_${lang}"
+done
+unset lang
+unset LANGS
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )"
+
+COMMON_DEPEND="x11-libs/wxGTK:${WX_GTK_VER}[X,opengl]
```
Comment (SoapGentoo) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: you're missing `setup-wxwidgets` somewhere (usually in `pkg_setup`):
https://devmanual.gentoo.org/eclass-reference/wxwidgets.eclass/index.html

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+	python? (
+		dev-python/wxpython:${WX_GTK_VER}[opengl,${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+	)
+	>=dev-libs/boost-1.56[nls,threads,python?]
+	github? (
+		libressl? ( dev-libs/libressl:0 )
+		!libressl? ( dev-libs/openssl:0 )
```
Comment (SoapGentoo) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: both these need `:0=`

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+REQUIRED_USE="
+	python? ( ${PYTHON_REQUIRED_USE} )"
+
+COMMON_DEPEND="x11-libs/wxGTK:${WX_GTK_VER}[X,opengl]
+	python? (
+		dev-python/wxpython:${WX_GTK_VER}[opengl,${PYTHON_USEDEP}]
+		${PYTHON_DEPS}
+	)
+	>=dev-libs/boost-1.56[nls,threads,python?]
```
Comment (SoapGentoo) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: also add `${PYTHON_USEDEP}`

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+	mkdir -p "${S}/common/system/" || die
+	mkdir -p "${S}/include/system/" || die
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.cpp" "${S}/common/system/libcontext.cpp" || die
+	cp "${WORKDIR}/${PN}-libcontext/libcontext.h" "${S}/include/system/libcontext.h" || die
+	# Path source to use new "built in" libcontext. Also patch libcontext.cpp to have correct include file.
+	# Path must be applied after new libcontext files have been copied to the kicad source directory.
+	eapply "${FILESDIR}/${PN}-4.0.4-boost-context.patch"
+
+	# remove all the non unix file endings
+	edos2unix $(find "${S}" -type f -name "*.desktop")
```
Comment (SoapGentoo) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: in general this should be done using the funky read while loop:
https://devmanual.gentoo.org/tools-reference/find/

File: sci-electronics/kicad/kicad-4.0.5.ebuild [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+	python? ( dev-lang/swig:0 )"
+RDEPEND="${COMMON_DEPEND}
+	sci-electronics/electronics-menu"
+
+pkg_setup() {
+	python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	xdg_src_prepare
```
Comment (SoapGentoo) [inline sci-electronics/kicad/kicad-4.0.5.ebuild]: if possible, replace this with `cmake-utils_src_prepare`

**Comment by gktrk** [comment]
Merged in https://github.com/gentoo/gentoo/compare/d88256e...88ce08f. Thanks!

**Pull request closed without merging**